Setup Wizard: Remove setup wizard

robertf4 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
This PR removes the Setup Wizard, which has been replaced by the Recommendations tab.

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->
N/A

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->
No

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

1. Start a new site and use the snippets plugin to add this, which would activate the Setup Wizard:
```
add_filter( 'jetpack_show_setup_wizard', '__return_true' );
add_filter( 'jetpack_pre_connection_prompt_helpers', '__return_true' );
```
2. Verify that the "Set up" option is not available in the left side nav bar.
3. Connect Jetpack and click around through various pages. Check the error log to make sure that there are no errors.


#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->
<!-- Guidelines: https://github.com/Automattic/jetpack/blob/master/docs/writing-a-good-changelog-entry.md -->
None needed (the setup wizard was tested only on Bluehost sites and was replaced by the Recommendations tab which is showing for all users)

robertf4 committed:
Remove setup wizard

jeherve reviewed (changes requested):
I would suggest that we keep those 2 files around for a few releases, to avoid Fatal errors during the plugin update, on sites with aggressive caching

```
projects/plugins/jetpack/class-jetpack-wizard-banner.php
projects/plugins/jetpack/_inc/lib/class-jetpack-wizard.php
```

We can keep them with only a comment at the top, something like this maybe?

```php
<?php
/**
 * Deprecated since 9.5.0
 *
 * @deprecated
 * @package automattic/jetpack
 */

// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
_deprecated_file( basename( __FILE__ ), 'jetpack-9.5.0' );
````

robertf4 committed:
Add PHP files with deprecated comment

robertf4 commented:
> I would suggest that we keep those 2 files around for a few releases, to avoid Fatal errors during the plugin update, on sites with aggressive caching

Sounds good to me, let me know if this isn't what you had in mind.

jeherve committed:
Remove the classes from the 2 files

We don't really need them since they are not called anywhere. We only want to keep the files around for any sites that may continue to require the files after the plugin update because of opcache.

jeherve reviewed (approved):
This should be good to merge now! 🥳 

(I updated the 2 files to only keep the deprecation warning, since the classes will not be called anywhere).
